In [66]:
# from google.colab import drive
# drive.mount('/content/drive')

In [67]:
# cd /content/drive/My Drive/University/SEM7/CS4622_Machine_Learning/Project_1/Fair_Classification

# Imports

In [68]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime
import calendar
import warnings
from math import sin, cos, sqrt, atan2, radians,asin

from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import GridSearchCV
from sklearn.ensemble import RandomForestRegressor
# import lightgbm as lgb
# import xgboost as xgb
from sklearn.metrics import roc_auc_score
# from hyperopt import hp, fmin, tpe, STATUS_OK, Trials
warnings.filterwarnings('ignore')
# from xgboost import XGBClassifier
from sklearn.metrics import accuracy_score,f1_score


# Load Data

In [69]:
# train=pd.read_csv("/content/drive/My Drive/University/SEM7/CS4622_Machine_Learning/Project_1/Fair_Classification/features/Train_FeatureEngineering_4_without_droping_any_feature.csv")
# test=pd.read_csv("/content/drive/My Drive/University/SEM7/CS4622_Machine_Learning/Project_1/Fair_Classification/features/Test_FeatureEngineering_4.csv")
# for other account
train=pd.read_csv("features/Train_FeatureEngineering_6_fill_null_with_0_changed_dtype.csv")
test=pd.read_csv("features/Test_FeatureEngineering_6_fill_null_with_0_changed_dtype.csv")

print("Shape of Training Data",train.shape)
print("Shape of Testing Data", test.shape)


Shape of Training Data (17060, 25)
Shape of Testing Data (8576, 24)


In [70]:
train.columns

Index(['tripid', 'additional_fare', 'duration', 'meter_waiting',
       'meter_waiting_fare', 'meter_waiting_till_pickup', 'pickup_time',
       'drop_time', 'pick_lat', 'pick_lon', 'drop_lat', 'drop_lon', 'fare',
       'label', 'additional_fare_is_outlier', 'meter_waiting_fare_is_outlier',
       'fare_is_outlier', 'pickup_date', 'pickup_day', 'pickup_hour',
       'pickup_day_of_week', 'pickup_month', 'pickup_year', 'is_outlier_loc',
       'trip_distance'],
      dtype='object')

In [71]:
test.columns

Index(['tripid', 'additional_fare', 'duration', 'meter_waiting',
       'meter_waiting_fare', 'meter_waiting_till_pickup', 'pickup_time',
       'drop_time', 'pick_lat', 'pick_lon', 'drop_lat', 'drop_lon', 'fare',
       'additional_fare_is_outlier', 'meter_waiting_fare_is_outlier',
       'fare_is_outlier', 'pickup_date', 'pickup_day', 'pickup_hour',
       'pickup_day_of_week', 'pickup_month', 'pickup_year', 'is_outlier_loc',
       'trip_distance'],
      dtype='object')

# Data_Preprocessing

In [72]:
# drop_columns_train=['tripid','pickup_time','drop_time','pick_lat','pick_lon','drop_lat','drop_lon','pickup_date','pickup_latitude_round3','pickup_longitude_round3','dropoff_latitude_round3','dropoff_longitude_round3']
# drop_columns_test=['tripid','pickup_time','drop_time','pick_lat','pick_lon','drop_lat','drop_lon','pickup_date','pickup_latitude_round3','pickup_longitude_round3','dropoff_latitude_round3','dropoff_longitude_round3','is_outlier_loc']

drop_columns_train=['tripid','pickup_time','drop_time','pickup_date']
drop_columns_test=['tripid','pickup_time','drop_time','pickup_date']

In [73]:
train_1=train.drop(drop_columns_train,axis=1)
test_1=test.drop(drop_columns_test,axis=1)
print("Shape of Training Data after dropping columns",train_1.shape)
print("Shape of Testing Data after dropping columns",test_1.shape)

Shape of Training Data after dropping columns (17060, 21)
Shape of Testing Data after dropping columns (8576, 20)


In [74]:
# def encodeDays(day_of_week):
#   day_dict={'Sunday':0,'Monday':1,'Tuesday':2,'Wednesday':3,'Thursday':4,'Friday':5,'Saturday':6}
#   return day_dict[day_of_week]
# def encodeLabel(label):
#   label_dict={'correct':1,'incorrect':0}
#   return label_dict[label]
# def clean_data(data,isTrain):
#   data['pickup_day_of_week']=data['pickup_day_of_week'].apply(lambda x:encodeDays(x))
#   if(isTrain):
#     data['label']=data['label'].apply(lambda x:encodeLabel(x))
#   return data

# train_1=clean_data(train_1,True)
# test_1=clean_data(test_1,False)
# print("Shape of Training Data after cleaning ",train_1.shape)
# print("Shape of Testing Data after cleaning", test_1.shape)

In [75]:
test_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8576 entries, 0 to 8575
Data columns (total 20 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   additional_fare                8576 non-null   float64
 1   duration                       8576 non-null   int64  
 2   meter_waiting                  8576 non-null   int64  
 3   meter_waiting_fare             8576 non-null   float64
 4   meter_waiting_till_pickup      8576 non-null   int64  
 5   pick_lat                       8576 non-null   float64
 6   pick_lon                       8576 non-null   float64
 7   drop_lat                       8576 non-null   float64
 8   drop_lon                       8576 non-null   float64
 9   fare                           8576 non-null   float64
 10  additional_fare_is_outlier     8576 non-null   int64  
 11  meter_waiting_fare_is_outlier  8576 non-null   int64  
 12  fare_is_outlier                8576 non-null   i

In [76]:
columns_titles = ['additional_fare','duration','meter_waiting','meter_waiting_fare','meter_waiting_till_pickup','pick_lat','pick_lon','drop_lat','drop_lon','fare','additional_fare_is_outlier','meter_waiting_fare_is_outlier','fare_is_outlier','pickup_day','pickup_hour','pickup_day_of_week','pickup_month','pickup_year','is_outlier_loc','trip_distance','label']
train_1=train_1.reindex(columns=columns_titles)
train_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17060 entries, 0 to 17059
Data columns (total 21 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   additional_fare                17060 non-null  float64
 1   duration                       17060 non-null  float64
 2   meter_waiting                  17060 non-null  float64
 3   meter_waiting_fare             17060 non-null  float64
 4   meter_waiting_till_pickup      17060 non-null  float64
 5   pick_lat                       17060 non-null  float64
 6   pick_lon                       17060 non-null  float64
 7   drop_lat                       17060 non-null  float64
 8   drop_lon                       17060 non-null  float64
 9   fare                           17060 non-null  float64
 10  additional_fare_is_outlier     17060 non-null  int64  
 11  meter_waiting_fare_is_outlier  17060 non-null  int64  
 12  fare_is_outlier                17060 non-null 

In [77]:
categorical_var = ['additional_fare_is_outlier','meter_waiting_fare_is_outlier','fare_is_outlier','pickup_day','pickup_hour','pickup_day_of_week','pickup_month','pickup_year','is_outlier_loc'] 
categorical_var_indices = [train_1.columns.get_loc(c) for c in categorical_var if c in train_1]
print('\nCategorical Variables indices : ',categorical_var_indices)


Categorical Variables indices :  [10, 11, 12, 13, 14, 15, 16, 17, 18]


In [78]:
# train_1=pd.get_dummies(train_1)
# test_1=pd.get_dummies(test_1)

# print("Shape of Training Data after One Hot Encoding",train_1.shape)
# print("Shape of Testing Data after One Hot Encoding",test_1.shape)

In [79]:
# print('Old size: %d' % len(train))
# train_1 = train_1.dropna()
# print('New size: %d' % len(train))

In [80]:
X=train_1.drop(['label'],axis=1)
y=train_1['label']

In [81]:
#split data into train and validation data
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.2)
print("Number of records in training data ",X_train.shape[0])
print("Number of records in validation data ",X_test.shape[0])

Number of records in training data  13648
Number of records in validation data  3412


# Classifiers

## XGBoost CLassifier

In [82]:
# model = XGBClassifier()
# model.fit(X_train, y_train)

## Linear Regression Classifier

In [83]:
# model = LinearRegression()
# model.fit(X_train,y_train)
# # y_pred=model.predict(X_test)
# # lm_rmse=np.sqrt(mean_squared_error(y_pred, y_test))
# # print("RMSE for Linear Regression is ",lm_rmse)


## LGBM Classifier

In [84]:
# estimator = lgb.LGBMClassifier(learning_rate = 0.125, metric = 'l1', 
#                         n_estimators = 20, num_leaves = 38)


# param_grid = {
#     'n_estimators': [x for x in range(24,40,2)],
#     'learning_rate': [0.10, 0.125, 0.15, 0.175, 0.2]}
# gridsearch = GridSearchCV(estimator, param_grid)

# gridsearch.fit(X_train, y_train,
#         eval_set = [(X_test, y_test)],
#         eval_metric = ['auc', 'binary_logloss'],
#         early_stopping_rounds = 5)

In [85]:
# gbm = lgb.LGBMClassifier(boosting_type='gbdt', class_weight=None,
#                                       colsample_bytree=1.0,
#                                       importance_type='split',
#                                       learning_rate=0.125, max_depth=-1,
#                                       metric='l1', min_child_samples=20,
#                                       min_child_weight=0.001,
#                                       min_split_gain=0.0, n_estimators=20,
#                                       n_jobs=-1, num_leaves=38, objective=None,
#                                       random_state=None, reg_alpha=0.0,
#                                       reg_lambda=0.0, silent=True,
#                                       subsample=1.0, subsample_for_bin=200000,
#                                       subsample_freq=0)


# gbm.fit(X, y,
#         eval_set=[(X_test, y_test)],
#         eval_metric=['auc', 'binary_logloss'],
# early_stopping_rounds=5)

In [86]:
# ax = lgb.plot_importance(gbm, height = 0.4, 
#                          max_num_features = 50, 
#                          xlim = (0,100), ylim = (0,23), 
#                          figsize = (10,6))
# plt.show()

In [87]:
# sorted(gbm.feature_importances_,reverse=True)

In [88]:

# # The code below aims to drop  to keep the features that are included in the most important features. 
# temp = 0 
# total = sum(gbm.feature_importances_)
# for feature in sorted(gbm.feature_importances_, reverse=True):
#     temp+=feature
#     print(feature)
#     if temp/total >= 0.85:
#         print(feature,temp/total) # stop when we 
#         break

In [89]:
# #The above means let go of all variables after PAY_AMT_5
# y_pred_prob = gbm.predict_proba(test_1)[:, 1]
# # auc_roc_0 = str(roc_auc_score(y_test, y_pred_prob)) # store AUC score without dimensionality reduction
# # print('AUC without dimensionality reduction: \n' + auc_roc_0)

In [90]:
X

additional_fare  duration  meter_waiting  meter_waiting_fare  \
0                 10.5     834.0           56.0              0.0000   
1                 10.5     791.0           47.0              0.0000   
2                 10.5    1087.0           80.0              0.0000   
3                 10.5     598.0          271.0             15.6638   
4                  0.0       0.0            0.0              0.0000   
...                ...       ...            ...                 ...   
17055             10.5     838.0           93.0              5.4219   
17056             10.5    2151.0          428.0              0.0000   
17057             10.5     263.0            9.0              0.0000   
17058             10.5     858.0          115.0              0.0000   
17059             10.5     262.0           21.0              1.2243   

       meter_waiting_till_pickup  pick_lat  pick_lon  drop_lat  drop_lon  \
0                           64.0   6.86252   79.8993   6.90330   79.8783   
1                          134.0   6.88589   79.8984   6.91373   79.8923   
2                           61.0   6.90839   79.8651   6.93669   79.9146   
3                           68.0   6.92570   79.8895   6.92748   79.8971   
4                            0.0   6.87441   79.8615   6.84478   79.9290   
...                          ...       ...       ...       ...       ...   
17055                      451.0   7.29073   80.6367   7.28891   80.6557   
17056                       39.0   6.90569   79.8516   6.95089   79.9389   
17057                      110.0   7.09210   79.9000   7.10135   79.9017   
17058                      317.0   6.94540   79.8768   6.93574   79.9010   
17059                       23.0   6.90257   79.9557   6.90823   79.9374   

         fare  additional_fare_is_outlier  meter_waiting_fare_is_outlier  \
0      270.32                           0                              0   
1      197.85                           0                              0   
2      301.64                           0                              0   
3       82.30                           0                              0   
4      358.39                           0                              0   
...       ...                         ...                            ...   
17055  198.26                           0                              0   
17056  581.23                           0                              0   
17057   76.20                           0                              0   
17058  133.31                           0                              0   
17059   98.57                           0                              0   

       fare_is_outlier  pickup_day  pickup_hour  pickup_day_of_week  \
0                    0           1            0                   5   
1                    0           1            0                   5   
2                    0           1            1                   5   
3                    0           1            2                   5   
4                    0           1            3                   5   
...                ...         ...          ...                 ...   
17055                0          31           22                   5   
17056                0          31           23                   5   
17057                0          31           23                   5   
17058                0          31           23                   5   
17059                0          31           23                   5   

       pickup_month  pickup_year  is_outlier_loc  trip_distance  
0                11         2019               0       3.164501  
1                11         2019               0       1.968540  
2                11         2019               0       3.917991  
3                11         2019               0       0.535588  
4                11         2019               0       5.062797  
...             ...          ...             ...            ...  
17055        

In [91]:
# #We can choose to drop the last 6 features from in our new model to reduce dimensionality, and thus save training time and space

# X = X_train.drop(['additional_fare','pickup_day_of_week','pickup_month','pickup_day'], axis=1)

## CatBoost

In [92]:
!pip install catboost

In [ ]:
from catboost import CatBoostClassifier

model = CatBoostClassifier(iterations=100000, 
                           cat_features=categorical_var_indices, 
                           task_type="GPU",
                           devices='0:1')
model.fit(X,
          y,
          verbose=True)




Learning rate set to 0.000466
0:	learn: 0.6925757	total: 86.1ms	remaining: 2h 23m 28s
1:	learn: 0.6920052	total: 170ms	remaining: 2h 21m 54s
2:	learn: 0.6915179	total: 249ms	remaining: 2h 18m 16s
3:	learn: 0.6909596	total: 328ms	remaining: 2h 16m 44s
4:	learn: 0.6904579	total: 406ms	remaining: 2h 15m 24s
5:	learn: 0.6899295	total: 481ms	remaining: 2h 13m 34s
6:	learn: 0.6894032	total: 551ms	remaining: 2h 11m 5s
7:	learn: 0.6888298	total: 608ms	remaining: 2h 6m 38s
8:	learn: 0.6882909	total: 679ms	remaining: 2h 5m 48s
9:	learn: 0.6878127	total: 745ms	remaining: 2h 4m 13s
10:	learn: 0.6873443	total: 812ms	remaining: 2h 2m 58s
11:	learn: 0.6867673	total: 876ms	remaining: 2h 1m 36s
12:	learn: 0.6862108	total: 937ms	remaining: 2h 10s
13:	learn: 0.6857314	total: 1.01s	remaining: 1h 59m 47s
14:	learn: 0.6851789	total: 1.07s	remaining: 1h 59m 4s
15:	learn: 0.6847060	total: 1.14s	remaining: 1h 58m 16s
16:	learn: 0.6842143	total: 1.19s	remaining: 1h 57m 5s
17:	learn: 0.6837381	total: 1.26s	remai

150:	learn: 0.6209670	total: 9.47s	remaining: 1h 44m 23s
151:	learn: 0.6205531	total: 9.53s	remaining: 1h 44m 23s
152:	learn: 0.6200661	total: 9.59s	remaining: 1h 44m 15s
153:	learn: 0.6195942	total: 9.65s	remaining: 1h 44m 14s
154:	learn: 0.6191903	total: 9.71s	remaining: 1h 44m 13s
155:	learn: 0.6187964	total: 9.76s	remaining: 1h 44m 8s
156:	learn: 0.6183941	total: 9.82s	remaining: 1h 44m 7s
157:	learn: 0.6179994	total: 9.89s	remaining: 1h 44m 9s
158:	learn: 0.6176183	total: 9.96s	remaining: 1h 44m 11s
159:	learn: 0.6171778	total: 10s	remaining: 1h 44m 10s
160:	learn: 0.6166661	total: 10.1s	remaining: 1h 44m 8s
161:	learn: 0.6161593	total: 10.1s	remaining: 1h 44m 6s
162:	learn: 0.6157700	total: 10.2s	remaining: 1h 44m 6s
163:	learn: 0.6153639	total: 10.3s	remaining: 1h 44m 4s
164:	learn: 0.6149768	total: 10.3s	remaining: 1h 44m 3s
165:	learn: 0.6145819	total: 10.4s	remaining: 1h 44m 4s
166:	learn: 0.6142035	total: 10.4s	remaining: 1h 44m 3s
167:	learn: 0.6138007	total: 10.5s	remainin

298:	learn: 0.5618509	total: 18.4s	remaining: 1h 42m 19s
299:	learn: 0.5614389	total: 18.5s	remaining: 1h 42m 19s
300:	learn: 0.5610090	total: 18.5s	remaining: 1h 42m 19s
301:	learn: 0.5606822	total: 18.6s	remaining: 1h 42m 20s
302:	learn: 0.5603490	total: 18.7s	remaining: 1h 42m 20s
303:	learn: 0.5599878	total: 18.7s	remaining: 1h 42m 20s
304:	learn: 0.5595412	total: 18.8s	remaining: 1h 42m 19s
305:	learn: 0.5592232	total: 18.8s	remaining: 1h 42m 10s
306:	learn: 0.5588462	total: 18.9s	remaining: 1h 42m 10s
307:	learn: 0.5584445	total: 18.9s	remaining: 1h 42m 9s
308:	learn: 0.5581187	total: 19s	remaining: 1h 42m 2s
309:	learn: 0.5577810	total: 19s	remaining: 1h 42m 2s
310:	learn: 0.5573864	total: 19.1s	remaining: 1h 42m 2s
311:	learn: 0.5569890	total: 19.2s	remaining: 1h 42m 1s
312:	learn: 0.5566338	total: 19.2s	remaining: 1h 42m 1s
313:	learn: 0.5561950	total: 19.3s	remaining: 1h 42m 1s
314:	learn: 0.5558713	total: 19.3s	remaining: 1h 42m 1s
315:	learn: 0.5554864	total: 19.4s	remainin

446:	learn: 0.5111536	total: 27.5s	remaining: 1h 42m 4s
447:	learn: 0.5108577	total: 27.6s	remaining: 1h 42m 4s
448:	learn: 0.5105841	total: 27.6s	remaining: 1h 42m 3s
449:	learn: 0.5102162	total: 27.7s	remaining: 1h 42m 3s
450:	learn: 0.5098368	total: 27.7s	remaining: 1h 42m 3s
451:	learn: 0.5094796	total: 27.8s	remaining: 1h 42m 4s
452:	learn: 0.5091131	total: 27.9s	remaining: 1h 42m 4s
453:	learn: 0.5088315	total: 27.9s	remaining: 1h 42m 5s
454:	learn: 0.5085503	total: 28s	remaining: 1h 42m 5s
455:	learn: 0.5082564	total: 28.1s	remaining: 1h 42m 5s
456:	learn: 0.5079242	total: 28.1s	remaining: 1h 42m 6s
457:	learn: 0.5076554	total: 28.2s	remaining: 1h 42m 7s
458:	learn: 0.5073723	total: 28.3s	remaining: 1h 42m 7s
459:	learn: 0.5070713	total: 28.3s	remaining: 1h 42m 6s
460:	learn: 0.5067224	total: 28.4s	remaining: 1h 42m 6s
461:	learn: 0.5064472	total: 28.4s	remaining: 1h 42m 5s
462:	learn: 0.5061073	total: 28.5s	remaining: 1h 42m 5s
463:	learn: 0.5057956	total: 28.6s	remaining: 1h 4

594:	learn: 0.4689436	total: 36.6s	remaining: 1h 41m 57s
595:	learn: 0.4686326	total: 36.7s	remaining: 1h 41m 57s
596:	learn: 0.4683517	total: 36.7s	remaining: 1h 41m 57s
597:	learn: 0.4681012	total: 36.8s	remaining: 1h 41m 57s
598:	learn: 0.4678720	total: 36.9s	remaining: 1h 41m 56s
599:	learn: 0.4676483	total: 36.9s	remaining: 1h 41m 57s
600:	learn: 0.4674248	total: 37s	remaining: 1h 41m 57s
601:	learn: 0.4671973	total: 37s	remaining: 1h 41m 57s
602:	learn: 0.4669682	total: 37.1s	remaining: 1h 41m 57s
603:	learn: 0.4666973	total: 37.2s	remaining: 1h 41m 57s
604:	learn: 0.4664067	total: 37.2s	remaining: 1h 41m 56s
605:	learn: 0.4661505	total: 37.3s	remaining: 1h 41m 56s
606:	learn: 0.4659237	total: 37.4s	remaining: 1h 41m 56s
607:	learn: 0.4656316	total: 37.4s	remaining: 1h 41m 56s
608:	learn: 0.4653923	total: 37.5s	remaining: 1h 41m 56s
609:	learn: 0.4651733	total: 37.5s	remaining: 1h 41m 56s
610:	learn: 0.4649294	total: 37.6s	remaining: 1h 41m 56s
611:	learn: 0.4646561	total: 37.7s	

742:	learn: 0.4324978	total: 45.7s	remaining: 1h 41m 48s
743:	learn: 0.4323124	total: 45.8s	remaining: 1h 41m 47s
744:	learn: 0.4321128	total: 45.8s	remaining: 1h 41m 47s
745:	learn: 0.4318268	total: 45.9s	remaining: 1h 41m 47s
746:	learn: 0.4316257	total: 45.9s	remaining: 1h 41m 44s
747:	learn: 0.4314273	total: 46s	remaining: 1h 41m 45s
748:	learn: 0.4312439	total: 46.1s	remaining: 1h 41m 45s
749:	learn: 0.4310609	total: 46.1s	remaining: 1h 41m 45s
750:	learn: 0.4307967	total: 46.2s	remaining: 1h 41m 45s
751:	learn: 0.4306104	total: 46.3s	remaining: 1h 41m 44s
752:	learn: 0.4304129	total: 46.3s	remaining: 1h 41m 44s
753:	learn: 0.4301539	total: 46.4s	remaining: 1h 41m 45s
754:	learn: 0.4298749	total: 46.4s	remaining: 1h 41m 45s
755:	learn: 0.4296943	total: 46.5s	remaining: 1h 41m 44s
756:	learn: 0.4294973	total: 46.6s	remaining: 1h 41m 44s
757:	learn: 0.4292739	total: 46.6s	remaining: 1h 41m 44s
758:	learn: 0.4290437	total: 46.7s	remaining: 1h 41m 44s
759:	learn: 0.4288579	total: 46.7

890:	learn: 0.4018717	total: 54.8s	remaining: 1h 41m 40s
891:	learn: 0.4016440	total: 54.9s	remaining: 1h 41m 40s
892:	learn: 0.4014157	total: 55s	remaining: 1h 41m 41s
893:	learn: 0.4012391	total: 55s	remaining: 1h 41m 41s
894:	learn: 0.4010287	total: 55.1s	remaining: 1h 41m 40s
895:	learn: 0.4007783	total: 55.2s	remaining: 1h 41m 40s
896:	learn: 0.4005883	total: 55.2s	remaining: 1h 41m 40s
897:	learn: 0.4003998	total: 55.3s	remaining: 1h 41m 41s
898:	learn: 0.4002459	total: 55.3s	remaining: 1h 41m 40s
899:	learn: 0.4000673	total: 55.4s	remaining: 1h 41m 41s
900:	learn: 0.3998898	total: 55.5s	remaining: 1h 41m 41s
901:	learn: 0.3996688	total: 55.5s	remaining: 1h 41m 41s
902:	learn: 0.3994686	total: 55.6s	remaining: 1h 41m 41s
903:	learn: 0.3993008	total: 55.7s	remaining: 1h 41m 41s
904:	learn: 0.3991006	total: 55.7s	remaining: 1h 41m 41s
905:	learn: 0.3989342	total: 55.8s	remaining: 1h 41m 41s
906:	learn: 0.3987404	total: 55.9s	remaining: 1h 41m 41s
907:	learn: 0.3985421	total: 55.9s	

1038:	learn: 0.3756912	total: 1m 3s	remaining: 1h 41m 21s
1039:	learn: 0.3755295	total: 1m 3s	remaining: 1h 41m 21s
1040:	learn: 0.3753921	total: 1m 3s	remaining: 1h 41m 21s
1041:	learn: 0.3752528	total: 1m 4s	remaining: 1h 41m 21s
1042:	learn: 0.3751034	total: 1m 4s	remaining: 1h 41m 22s
1043:	learn: 0.3749274	total: 1m 4s	remaining: 1h 41m 22s
1044:	learn: 0.3747276	total: 1m 4s	remaining: 1h 41m 23s
1045:	learn: 0.3745679	total: 1m 4s	remaining: 1h 41m 23s
1046:	learn: 0.3743923	total: 1m 4s	remaining: 1h 41m 23s
1047:	learn: 0.3742517	total: 1m 4s	remaining: 1h 41m 23s
1048:	learn: 0.3740680	total: 1m 4s	remaining: 1h 41m 24s
1049:	learn: 0.3738943	total: 1m 4s	remaining: 1h 41m 24s
1050:	learn: 0.3737588	total: 1m 4s	remaining: 1h 41m 24s
1051:	learn: 0.3735676	total: 1m 4s	remaining: 1h 41m 24s
1052:	learn: 0.3734321	total: 1m 4s	remaining: 1h 41m 24s
1053:	learn: 0.3732635	total: 1m 4s	remaining: 1h 41m 23s
1054:	learn: 0.3730813	total: 1m 4s	remaining: 1h 41m 23s
1055:	learn: 0

1182:	learn: 0.3537643	total: 1m 12s	remaining: 1h 41m 13s
1183:	learn: 0.3535924	total: 1m 12s	remaining: 1h 41m 13s
1184:	learn: 0.3534057	total: 1m 12s	remaining: 1h 41m 12s
1185:	learn: 0.3532227	total: 1m 12s	remaining: 1h 41m 12s
1186:	learn: 0.3530635	total: 1m 12s	remaining: 1h 41m 12s
1187:	learn: 0.3529515	total: 1m 13s	remaining: 1h 41m 12s
1188:	learn: 0.3528015	total: 1m 13s	remaining: 1h 41m 12s
1189:	learn: 0.3526891	total: 1m 13s	remaining: 1h 41m 11s
1190:	learn: 0.3525697	total: 1m 13s	remaining: 1h 41m 11s
1191:	learn: 0.3524143	total: 1m 13s	remaining: 1h 41m 11s
1192:	learn: 0.3523097	total: 1m 13s	remaining: 1h 41m 9s
1193:	learn: 0.3521381	total: 1m 13s	remaining: 1h 41m 9s
1194:	learn: 0.3520157	total: 1m 13s	remaining: 1h 41m 9s
1195:	learn: 0.3518884	total: 1m 13s	remaining: 1h 41m 9s
1196:	learn: 0.3517033	total: 1m 13s	remaining: 1h 41m 9s
1197:	learn: 0.3515329	total: 1m 13s	remaining: 1h 41m 8s
1198:	learn: 0.3513536	total: 1m 13s	remaining: 1h 41m 8s
1199

1326:	learn: 0.3345367	total: 1m 21s	remaining: 1h 40m 56s
1327:	learn: 0.3343780	total: 1m 21s	remaining: 1h 40m 56s
1328:	learn: 0.3342189	total: 1m 21s	remaining: 1h 40m 56s
1329:	learn: 0.3340813	total: 1m 21s	remaining: 1h 40m 56s
1330:	learn: 0.3339443	total: 1m 21s	remaining: 1h 40m 56s
1331:	learn: 0.3338448	total: 1m 21s	remaining: 1h 40m 56s
1332:	learn: 0.3336989	total: 1m 21s	remaining: 1h 40m 56s
1333:	learn: 0.3335575	total: 1m 21s	remaining: 1h 40m 56s
1334:	learn: 0.3334170	total: 1m 21s	remaining: 1h 40m 56s
1335:	learn: 0.3333207	total: 1m 22s	remaining: 1h 40m 56s
1336:	learn: 0.3331763	total: 1m 22s	remaining: 1h 40m 56s
1337:	learn: 0.3330838	total: 1m 22s	remaining: 1h 40m 56s
1338:	learn: 0.3329246	total: 1m 22s	remaining: 1h 40m 56s
1339:	learn: 0.3328342	total: 1m 22s	remaining: 1h 40m 56s
1340:	learn: 0.3326762	total: 1m 22s	remaining: 1h 40m 56s
1341:	learn: 0.3325194	total: 1m 22s	remaining: 1h 40m 56s
1342:	learn: 0.3324198	total: 1m 22s	remaining: 1h 40m 5

1466:	learn: 0.3181246	total: 1m 30s	remaining: 1h 40m 50s
1467:	learn: 0.3180483	total: 1m 30s	remaining: 1h 40m 50s
1468:	learn: 0.3178991	total: 1m 30s	remaining: 1h 40m 50s
1469:	learn: 0.3178040	total: 1m 30s	remaining: 1h 40m 50s
1470:	learn: 0.3177114	total: 1m 30s	remaining: 1h 40m 50s
1471:	learn: 0.3176274	total: 1m 30s	remaining: 1h 40m 50s
1472:	learn: 0.3175361	total: 1m 30s	remaining: 1h 40m 50s
1473:	learn: 0.3174477	total: 1m 30s	remaining: 1h 40m 50s
1474:	learn: 0.3173274	total: 1m 30s	remaining: 1h 40m 50s
1475:	learn: 0.3171940	total: 1m 30s	remaining: 1h 40m 49s
1476:	learn: 0.3170665	total: 1m 30s	remaining: 1h 40m 49s
1477:	learn: 0.3169417	total: 1m 30s	remaining: 1h 40m 49s
1478:	learn: 0.3168149	total: 1m 30s	remaining: 1h 40m 49s
1479:	learn: 0.3167387	total: 1m 30s	remaining: 1h 40m 49s
1480:	learn: 0.3166414	total: 1m 30s	remaining: 1h 40m 49s
1481:	learn: 0.3165202	total: 1m 31s	remaining: 1h 40m 49s
1482:	learn: 0.3164455	total: 1m 31s	remaining: 1h 40m 4

1606:	learn: 0.3029619	total: 1m 38s	remaining: 1h 40m 40s
1607:	learn: 0.3028732	total: 1m 38s	remaining: 1h 40m 40s
1608:	learn: 0.3027910	total: 1m 38s	remaining: 1h 40m 40s
1609:	learn: 0.3027220	total: 1m 38s	remaining: 1h 40m 40s
1610:	learn: 0.3026207	total: 1m 38s	remaining: 1h 40m 40s
1611:	learn: 0.3025002	total: 1m 38s	remaining: 1h 40m 40s
1612:	learn: 0.3024108	total: 1m 39s	remaining: 1h 40m 40s
1613:	learn: 0.3023362	total: 1m 39s	remaining: 1h 40m 40s
1614:	learn: 0.3022207	total: 1m 39s	remaining: 1h 40m 40s
1615:	learn: 0.3021070	total: 1m 39s	remaining: 1h 40m 40s
1616:	learn: 0.3020177	total: 1m 39s	remaining: 1h 40m 40s
1617:	learn: 0.3019443	total: 1m 39s	remaining: 1h 40m 40s
1618:	learn: 0.3018257	total: 1m 39s	remaining: 1h 40m 40s
1619:	learn: 0.3017295	total: 1m 39s	remaining: 1h 40m 40s
1620:	learn: 0.3016063	total: 1m 39s	remaining: 1h 40m 40s
1621:	learn: 0.3014923	total: 1m 39s	remaining: 1h 40m 40s
1622:	learn: 0.3014076	total: 1m 39s	remaining: 1h 40m 4

1746:	learn: 0.2899945	total: 1m 47s	remaining: 1h 40m 36s
1747:	learn: 0.2899007	total: 1m 47s	remaining: 1h 40m 36s
1748:	learn: 0.2897963	total: 1m 47s	remaining: 1h 40m 36s
1749:	learn: 0.2896891	total: 1m 47s	remaining: 1h 40m 36s
1750:	learn: 0.2895872	total: 1m 47s	remaining: 1h 40m 36s
1751:	learn: 0.2895097	total: 1m 47s	remaining: 1h 40m 37s
1752:	learn: 0.2894510	total: 1m 47s	remaining: 1h 40m 37s
1753:	learn: 0.2893583	total: 1m 47s	remaining: 1h 40m 37s
1754:	learn: 0.2892935	total: 1m 47s	remaining: 1h 40m 37s
1755:	learn: 0.2892180	total: 1m 47s	remaining: 1h 40m 37s
1756:	learn: 0.2891171	total: 1m 47s	remaining: 1h 40m 37s
1757:	learn: 0.2890593	total: 1m 48s	remaining: 1h 40m 37s
1758:	learn: 0.2889695	total: 1m 48s	remaining: 1h 40m 37s
1759:	learn: 0.2889128	total: 1m 48s	remaining: 1h 40m 37s
1760:	learn: 0.2888561	total: 1m 48s	remaining: 1h 40m 37s
1761:	learn: 0.2887564	total: 1m 48s	remaining: 1h 40m 37s
1762:	learn: 0.2886463	total: 1m 48s	remaining: 1h 40m 3

1886:	learn: 0.2788473	total: 1m 56s	remaining: 1h 40m 34s
1887:	learn: 0.2787894	total: 1m 56s	remaining: 1h 40m 34s
1888:	learn: 0.2787394	total: 1m 56s	remaining: 1h 40m 34s
1889:	learn: 0.2786854	total: 1m 56s	remaining: 1h 40m 34s
1890:	learn: 0.2786365	total: 1m 56s	remaining: 1h 40m 34s
1891:	learn: 0.2785518	total: 1m 56s	remaining: 1h 40m 34s
1892:	learn: 0.2784551	total: 1m 56s	remaining: 1h 40m 34s
1893:	learn: 0.2783697	total: 1m 56s	remaining: 1h 40m 34s
1894:	learn: 0.2782846	total: 1m 56s	remaining: 1h 40m 34s
1895:	learn: 0.2781816	total: 1m 56s	remaining: 1h 40m 34s
1896:	learn: 0.2781012	total: 1m 56s	remaining: 1h 40m 34s
1897:	learn: 0.2780252	total: 1m 56s	remaining: 1h 40m 34s
1898:	learn: 0.2779436	total: 1m 56s	remaining: 1h 40m 34s
1899:	learn: 0.2778535	total: 1m 56s	remaining: 1h 40m 34s
1900:	learn: 0.2777615	total: 1m 56s	remaining: 1h 40m 34s
1901:	learn: 0.2776909	total: 1m 57s	remaining: 1h 40m 34s
1902:	learn: 0.2776009	total: 1m 57s	remaining: 1h 40m 3

2028:	learn: 0.2688628	total: 2m 5s	remaining: 1h 40m 35s
2029:	learn: 0.2687856	total: 2m 5s	remaining: 1h 40m 36s
2030:	learn: 0.2687150	total: 2m 5s	remaining: 1h 40m 36s
2031:	learn: 0.2686249	total: 2m 5s	remaining: 1h 40m 35s
2032:	learn: 0.2685346	total: 2m 5s	remaining: 1h 40m 35s
2033:	learn: 0.2684561	total: 2m 5s	remaining: 1h 40m 35s
2034:	learn: 0.2684078	total: 2m 5s	remaining: 1h 40m 36s
2035:	learn: 0.2683356	total: 2m 5s	remaining: 1h 40m 36s
2036:	learn: 0.2682626	total: 2m 5s	remaining: 1h 40m 36s
2037:	learn: 0.2682223	total: 2m 5s	remaining: 1h 40m 36s
2038:	learn: 0.2681590	total: 2m 5s	remaining: 1h 40m 36s
2039:	learn: 0.2680782	total: 2m 5s	remaining: 1h 40m 36s
2040:	learn: 0.2680393	total: 2m 5s	remaining: 1h 40m 36s
2041:	learn: 0.2679827	total: 2m 5s	remaining: 1h 40m 36s
2042:	learn: 0.2679006	total: 2m 5s	remaining: 1h 40m 36s
2043:	learn: 0.2678319	total: 2m 5s	remaining: 1h 40m 36s
2044:	learn: 0.2677889	total: 2m 6s	remaining: 1h 40m 36s
2045:	learn: 0

2170:	learn: 0.2600152	total: 2m 13s	remaining: 1h 40m 34s
2171:	learn: 0.2599787	total: 2m 13s	remaining: 1h 40m 34s
2172:	learn: 0.2599096	total: 2m 14s	remaining: 1h 40m 34s
2173:	learn: 0.2598666	total: 2m 14s	remaining: 1h 40m 34s
2174:	learn: 0.2597948	total: 2m 14s	remaining: 1h 40m 34s
2175:	learn: 0.2597216	total: 2m 14s	remaining: 1h 40m 34s
2176:	learn: 0.2596409	total: 2m 14s	remaining: 1h 40m 34s
2177:	learn: 0.2596016	total: 2m 14s	remaining: 1h 40m 34s
2178:	learn: 0.2595395	total: 2m 14s	remaining: 1h 40m 34s
2179:	learn: 0.2595004	total: 2m 14s	remaining: 1h 40m 34s
2180:	learn: 0.2594620	total: 2m 14s	remaining: 1h 40m 34s
2181:	learn: 0.2593902	total: 2m 14s	remaining: 1h 40m 34s
2182:	learn: 0.2593546	total: 2m 14s	remaining: 1h 40m 33s
2183:	learn: 0.2593138	total: 2m 14s	remaining: 1h 40m 33s
2184:	learn: 0.2592451	total: 2m 14s	remaining: 1h 40m 33s
2185:	learn: 0.2591671	total: 2m 14s	remaining: 1h 40m 33s
2186:	learn: 0.2591053	total: 2m 14s	remaining: 1h 40m 3

2312:	learn: 0.2522090	total: 2m 22s	remaining: 1h 40m 32s
2313:	learn: 0.2521502	total: 2m 22s	remaining: 1h 40m 32s
2314:	learn: 0.2521042	total: 2m 22s	remaining: 1h 40m 32s
2315:	learn: 0.2520646	total: 2m 23s	remaining: 1h 40m 32s
2316:	learn: 0.2520230	total: 2m 23s	remaining: 1h 40m 32s
2317:	learn: 0.2519838	total: 2m 23s	remaining: 1h 40m 33s
2318:	learn: 0.2519589	total: 2m 23s	remaining: 1h 40m 32s
2319:	learn: 0.2519316	total: 2m 23s	remaining: 1h 40m 32s
2320:	learn: 0.2518733	total: 2m 23s	remaining: 1h 40m 32s
2321:	learn: 0.2518179	total: 2m 23s	remaining: 1h 40m 32s
2322:	learn: 0.2517472	total: 2m 23s	remaining: 1h 40m 32s
2323:	learn: 0.2516934	total: 2m 23s	remaining: 1h 40m 32s
2324:	learn: 0.2516280	total: 2m 23s	remaining: 1h 40m 32s
2325:	learn: 0.2515862	total: 2m 23s	remaining: 1h 40m 32s
2326:	learn: 0.2515110	total: 2m 23s	remaining: 1h 40m 32s
2327:	learn: 0.2514801	total: 2m 23s	remaining: 1h 40m 32s
2328:	learn: 0.2514403	total: 2m 23s	remaining: 1h 40m 3

2452:	learn: 0.2455202	total: 2m 31s	remaining: 1h 40m 33s
2453:	learn: 0.2454680	total: 2m 31s	remaining: 1h 40m 34s
2454:	learn: 0.2454225	total: 2m 31s	remaining: 1h 40m 34s
2455:	learn: 0.2453646	total: 2m 31s	remaining: 1h 40m 34s
2456:	learn: 0.2453234	total: 2m 32s	remaining: 1h 40m 34s
2457:	learn: 0.2452663	total: 2m 32s	remaining: 1h 40m 34s
2458:	learn: 0.2452109	total: 2m 32s	remaining: 1h 40m 34s
2459:	learn: 0.2451794	total: 2m 32s	remaining: 1h 40m 34s
2460:	learn: 0.2451380	total: 2m 32s	remaining: 1h 40m 34s
2461:	learn: 0.2450759	total: 2m 32s	remaining: 1h 40m 34s
2462:	learn: 0.2450248	total: 2m 32s	remaining: 1h 40m 34s
2463:	learn: 0.2449890	total: 2m 32s	remaining: 1h 40m 34s
2464:	learn: 0.2449234	total: 2m 32s	remaining: 1h 40m 34s
2465:	learn: 0.2448705	total: 2m 32s	remaining: 1h 40m 34s
2466:	learn: 0.2448399	total: 2m 32s	remaining: 1h 40m 34s
2467:	learn: 0.2447758	total: 2m 32s	remaining: 1h 40m 34s
2468:	learn: 0.2447414	total: 2m 32s	remaining: 1h 40m 3

2591:	learn: 0.2394721	total: 2m 40s	remaining: 1h 40m 34s
2592:	learn: 0.2394384	total: 2m 40s	remaining: 1h 40m 34s
2593:	learn: 0.2394094	total: 2m 40s	remaining: 1h 40m 34s
2594:	learn: 0.2393792	total: 2m 40s	remaining: 1h 40m 34s
2595:	learn: 0.2393493	total: 2m 40s	remaining: 1h 40m 34s
2596:	learn: 0.2393040	total: 2m 40s	remaining: 1h 40m 34s
2597:	learn: 0.2392539	total: 2m 40s	remaining: 1h 40m 34s
2598:	learn: 0.2392035	total: 2m 41s	remaining: 1h 40m 34s
2599:	learn: 0.2391435	total: 2m 41s	remaining: 1h 40m 34s
2600:	learn: 0.2390848	total: 2m 41s	remaining: 1h 40m 34s
2601:	learn: 0.2390417	total: 2m 41s	remaining: 1h 40m 34s
2602:	learn: 0.2389937	total: 2m 41s	remaining: 1h 40m 34s
2603:	learn: 0.2389506	total: 2m 41s	remaining: 1h 40m 34s
2604:	learn: 0.2389267	total: 2m 41s	remaining: 1h 40m 34s
2605:	learn: 0.2388792	total: 2m 41s	remaining: 1h 40m 34s
2606:	learn: 0.2388250	total: 2m 41s	remaining: 1h 40m 34s
2607:	learn: 0.2387949	total: 2m 41s	remaining: 1h 40m 3

2731:	learn: 0.2341545	total: 2m 49s	remaining: 1h 40m 27s
2732:	learn: 0.2341327	total: 2m 49s	remaining: 1h 40m 27s
2733:	learn: 0.2341123	total: 2m 49s	remaining: 1h 40m 27s
2734:	learn: 0.2340840	total: 2m 49s	remaining: 1h 40m 27s
2735:	learn: 0.2340527	total: 2m 49s	remaining: 1h 40m 27s
2736:	learn: 0.2340214	total: 2m 49s	remaining: 1h 40m 27s
2737:	learn: 0.2339809	total: 2m 49s	remaining: 1h 40m 27s
2738:	learn: 0.2339484	total: 2m 49s	remaining: 1h 40m 27s
2739:	learn: 0.2339225	total: 2m 49s	remaining: 1h 40m 27s
2740:	learn: 0.2338863	total: 2m 49s	remaining: 1h 40m 27s
2741:	learn: 0.2338645	total: 2m 49s	remaining: 1h 40m 27s
2742:	learn: 0.2338455	total: 2m 49s	remaining: 1h 40m 27s
2743:	learn: 0.2338069	total: 2m 50s	remaining: 1h 40m 27s
2744:	learn: 0.2337560	total: 2m 50s	remaining: 1h 40m 27s
2745:	learn: 0.2337252	total: 2m 50s	remaining: 1h 40m 27s
2746:	learn: 0.2336864	total: 2m 50s	remaining: 1h 40m 27s
2747:	learn: 0.2336490	total: 2m 50s	remaining: 1h 40m 2

2871:	learn: 0.2293892	total: 2m 58s	remaining: 1h 40m 20s
2872:	learn: 0.2293433	total: 2m 58s	remaining: 1h 40m 20s
2873:	learn: 0.2292935	total: 2m 58s	remaining: 1h 40m 20s
2874:	learn: 0.2292595	total: 2m 58s	remaining: 1h 40m 20s
2875:	learn: 0.2292336	total: 2m 58s	remaining: 1h 40m 20s
2876:	learn: 0.2291898	total: 2m 58s	remaining: 1h 40m 20s
2877:	learn: 0.2291518	total: 2m 58s	remaining: 1h 40m 20s
2878:	learn: 0.2291250	total: 2m 58s	remaining: 1h 40m 20s
2879:	learn: 0.2290812	total: 2m 58s	remaining: 1h 40m 20s
2880:	learn: 0.2290432	total: 2m 58s	remaining: 1h 40m 20s
2881:	learn: 0.2290082	total: 2m 58s	remaining: 1h 40m 20s
2882:	learn: 0.2289855	total: 2m 58s	remaining: 1h 40m 19s
2883:	learn: 0.2289660	total: 2m 58s	remaining: 1h 40m 20s
2884:	learn: 0.2289385	total: 2m 58s	remaining: 1h 40m 19s
2885:	learn: 0.2288979	total: 2m 58s	remaining: 1h 40m 19s
2886:	learn: 0.2288485	total: 2m 58s	remaining: 1h 40m 19s
2887:	learn: 0.2288303	total: 2m 59s	remaining: 1h 40m 1

3015:	learn: 0.2250040	total: 3m 6s	remaining: 1h 40m 10s
3016:	learn: 0.2249749	total: 3m 6s	remaining: 1h 40m 10s
3017:	learn: 0.2249553	total: 3m 7s	remaining: 1h 40m 10s
3018:	learn: 0.2249170	total: 3m 7s	remaining: 1h 40m 10s
3019:	learn: 0.2249012	total: 3m 7s	remaining: 1h 40m 10s
3020:	learn: 0.2248808	total: 3m 7s	remaining: 1h 40m 10s
3021:	learn: 0.2248414	total: 3m 7s	remaining: 1h 40m 10s
3022:	learn: 0.2248027	total: 3m 7s	remaining: 1h 40m 10s
3023:	learn: 0.2247826	total: 3m 7s	remaining: 1h 40m 10s
3024:	learn: 0.2247618	total: 3m 7s	remaining: 1h 40m 10s
3025:	learn: 0.2247397	total: 3m 7s	remaining: 1h 40m 10s
3026:	learn: 0.2246997	total: 3m 7s	remaining: 1h 40m 10s
3027:	learn: 0.2246683	total: 3m 7s	remaining: 1h 40m 10s
3028:	learn: 0.2246343	total: 3m 7s	remaining: 1h 40m 10s
3029:	learn: 0.2245969	total: 3m 7s	remaining: 1h 40m 10s
3030:	learn: 0.2245676	total: 3m 7s	remaining: 1h 40m 10s
3031:	learn: 0.2245406	total: 3m 7s	remaining: 1h 40m 10s
3032:	learn: 0

3159:	learn: 0.2211365	total: 3m 15s	remaining: 1h 40m 3s
3160:	learn: 0.2210975	total: 3m 15s	remaining: 1h 40m 3s
3161:	learn: 0.2210773	total: 3m 16s	remaining: 1h 40m 3s
3162:	learn: 0.2210575	total: 3m 16s	remaining: 1h 40m 3s
3163:	learn: 0.2210244	total: 3m 16s	remaining: 1h 40m 3s
3164:	learn: 0.2209821	total: 3m 16s	remaining: 1h 40m 3s
3165:	learn: 0.2209495	total: 3m 16s	remaining: 1h 40m 3s
3166:	learn: 0.2209264	total: 3m 16s	remaining: 1h 40m 3s
3167:	learn: 0.2208943	total: 3m 16s	remaining: 1h 40m 3s
3168:	learn: 0.2208591	total: 3m 16s	remaining: 1h 40m 3s
3169:	learn: 0.2208478	total: 3m 16s	remaining: 1h 40m 3s
3170:	learn: 0.2208267	total: 3m 16s	remaining: 1h 40m 3s
3171:	learn: 0.2207855	total: 3m 16s	remaining: 1h 40m 3s
3172:	learn: 0.2207537	total: 3m 16s	remaining: 1h 40m 3s
3173:	learn: 0.2207291	total: 3m 16s	remaining: 1h 40m 2s
3174:	learn: 0.2206984	total: 3m 16s	remaining: 1h 40m 2s
3175:	learn: 0.2206678	total: 3m 16s	remaining: 1h 40m 2s
3176:	learn: 0

3302:	learn: 0.2176029	total: 3m 24s	remaining: 1h 39m 54s
3303:	learn: 0.2175867	total: 3m 24s	remaining: 1h 39m 54s
3304:	learn: 0.2175610	total: 3m 24s	remaining: 1h 39m 53s
3305:	learn: 0.2175436	total: 3m 24s	remaining: 1h 39m 53s
3306:	learn: 0.2175177	total: 3m 24s	remaining: 1h 39m 53s
3307:	learn: 0.2174954	total: 3m 25s	remaining: 1h 39m 53s
3308:	learn: 0.2174784	total: 3m 25s	remaining: 1h 39m 53s
3309:	learn: 0.2174634	total: 3m 25s	remaining: 1h 39m 53s
3310:	learn: 0.2174428	total: 3m 25s	remaining: 1h 39m 53s
3311:	learn: 0.2174076	total: 3m 25s	remaining: 1h 39m 53s
3312:	learn: 0.2173918	total: 3m 25s	remaining: 1h 39m 53s
3313:	learn: 0.2173678	total: 3m 25s	remaining: 1h 39m 53s
3314:	learn: 0.2173512	total: 3m 25s	remaining: 1h 39m 53s
3315:	learn: 0.2173248	total: 3m 25s	remaining: 1h 39m 52s
3316:	learn: 0.2173101	total: 3m 25s	remaining: 1h 39m 52s
3317:	learn: 0.2172965	total: 3m 25s	remaining: 1h 39m 52s
3318:	learn: 0.2172850	total: 3m 25s	remaining: 1h 39m 5

3442:	learn: 0.2144062	total: 3m 33s	remaining: 1h 39m 44s
3443:	learn: 0.2143885	total: 3m 33s	remaining: 1h 39m 44s
3444:	learn: 0.2143664	total: 3m 33s	remaining: 1h 39m 44s
3445:	learn: 0.2143331	total: 3m 33s	remaining: 1h 39m 44s
3446:	learn: 0.2143087	total: 3m 33s	remaining: 1h 39m 44s
3447:	learn: 0.2142797	total: 3m 33s	remaining: 1h 39m 44s
3448:	learn: 0.2142626	total: 3m 33s	remaining: 1h 39m 44s
3449:	learn: 0.2142406	total: 3m 33s	remaining: 1h 39m 44s
3450:	learn: 0.2142224	total: 3m 33s	remaining: 1h 39m 44s
3451:	learn: 0.2142085	total: 3m 33s	remaining: 1h 39m 44s
3452:	learn: 0.2141916	total: 3m 34s	remaining: 1h 39m 44s
3453:	learn: 0.2141619	total: 3m 34s	remaining: 1h 39m 44s
3454:	learn: 0.2141494	total: 3m 34s	remaining: 1h 39m 44s
3455:	learn: 0.2141184	total: 3m 34s	remaining: 1h 39m 44s
3456:	learn: 0.2141036	total: 3m 34s	remaining: 1h 39m 43s
3457:	learn: 0.2140896	total: 3m 34s	remaining: 1h 39m 44s
3458:	learn: 0.2140765	total: 3m 34s	remaining: 1h 39m 4

3582:	learn: 0.2114936	total: 3m 42s	remaining: 1h 39m 36s
3583:	learn: 0.2114844	total: 3m 42s	remaining: 1h 39m 36s
3584:	learn: 0.2114705	total: 3m 42s	remaining: 1h 39m 36s
3585:	learn: 0.2114427	total: 3m 42s	remaining: 1h 39m 36s
3586:	learn: 0.2114096	total: 3m 42s	remaining: 1h 39m 36s
3587:	learn: 0.2113921	total: 3m 42s	remaining: 1h 39m 36s
3588:	learn: 0.2113634	total: 3m 42s	remaining: 1h 39m 35s
3589:	learn: 0.2113501	total: 3m 42s	remaining: 1h 39m 35s
3590:	learn: 0.2113343	total: 3m 42s	remaining: 1h 39m 35s
3591:	learn: 0.2113220	total: 3m 42s	remaining: 1h 39m 35s
3592:	learn: 0.2113117	total: 3m 42s	remaining: 1h 39m 35s
3593:	learn: 0.2112995	total: 3m 42s	remaining: 1h 39m 35s
3594:	learn: 0.2112769	total: 3m 42s	remaining: 1h 39m 35s
3595:	learn: 0.2112624	total: 3m 42s	remaining: 1h 39m 35s
3596:	learn: 0.2112373	total: 3m 42s	remaining: 1h 39m 35s
3597:	learn: 0.2112293	total: 3m 43s	remaining: 1h 39m 35s
3598:	learn: 0.2112091	total: 3m 43s	remaining: 1h 39m 3

3722:	learn: 0.2088578	total: 3m 50s	remaining: 1h 39m 26s
3723:	learn: 0.2088330	total: 3m 50s	remaining: 1h 39m 26s
3724:	learn: 0.2088260	total: 3m 50s	remaining: 1h 39m 26s
3725:	learn: 0.2088148	total: 3m 50s	remaining: 1h 39m 26s
3726:	learn: 0.2088034	total: 3m 50s	remaining: 1h 39m 26s
3727:	learn: 0.2087765	total: 3m 51s	remaining: 1h 39m 26s
3728:	learn: 0.2087581	total: 3m 51s	remaining: 1h 39m 26s
3729:	learn: 0.2087464	total: 3m 51s	remaining: 1h 39m 25s
3730:	learn: 0.2087334	total: 3m 51s	remaining: 1h 39m 25s
3731:	learn: 0.2087022	total: 3m 51s	remaining: 1h 39m 25s
3732:	learn: 0.2086831	total: 3m 51s	remaining: 1h 39m 25s
3733:	learn: 0.2086667	total: 3m 51s	remaining: 1h 39m 25s
3734:	learn: 0.2086437	total: 3m 51s	remaining: 1h 39m 25s
3735:	learn: 0.2086152	total: 3m 51s	remaining: 1h 39m 25s
3736:	learn: 0.2086002	total: 3m 51s	remaining: 1h 39m 25s
3737:	learn: 0.2085863	total: 3m 51s	remaining: 1h 39m 25s
3738:	learn: 0.2085623	total: 3m 51s	remaining: 1h 39m 2

3862:	learn: 0.2063338	total: 3m 59s	remaining: 1h 39m 18s
3863:	learn: 0.2063217	total: 3m 59s	remaining: 1h 39m 18s
3864:	learn: 0.2062979	total: 3m 59s	remaining: 1h 39m 18s
3865:	learn: 0.2062742	total: 3m 59s	remaining: 1h 39m 18s
3866:	learn: 0.2062558	total: 3m 59s	remaining: 1h 39m 18s
3867:	learn: 0.2062389	total: 3m 59s	remaining: 1h 39m 18s
3868:	learn: 0.2062101	total: 3m 59s	remaining: 1h 39m 17s
3869:	learn: 0.2061845	total: 3m 59s	remaining: 1h 39m 17s
3870:	learn: 0.2061711	total: 3m 59s	remaining: 1h 39m 17s
3871:	learn: 0.2061590	total: 3m 59s	remaining: 1h 39m 17s
3872:	learn: 0.2061330	total: 4m	remaining: 1h 39m 17s
3873:	learn: 0.2061171	total: 4m	remaining: 1h 39m 17s
3874:	learn: 0.2060938	total: 4m	remaining: 1h 39m 17s
3875:	learn: 0.2060839	total: 4m	remaining: 1h 39m 17s
3876:	learn: 0.2060716	total: 4m	remaining: 1h 39m 17s
3877:	learn: 0.2060567	total: 4m	remaining: 1h 39m 17s
3878:	learn: 0.2060439	total: 4m	remaining: 1h 39m 17s
3879:	learn: 0.2060351	to

4006:	learn: 0.2039771	total: 4m 8s	remaining: 1h 39m 9s
4007:	learn: 0.2039605	total: 4m 8s	remaining: 1h 39m 9s
4008:	learn: 0.2039338	total: 4m 8s	remaining: 1h 39m 9s
4009:	learn: 0.2039267	total: 4m 8s	remaining: 1h 39m 9s
4010:	learn: 0.2039143	total: 4m 8s	remaining: 1h 39m 9s
4011:	learn: 0.2038935	total: 4m 8s	remaining: 1h 39m 9s
4012:	learn: 0.2038766	total: 4m 8s	remaining: 1h 39m 9s
4013:	learn: 0.2038535	total: 4m 8s	remaining: 1h 39m 9s
4014:	learn: 0.2038334	total: 4m 8s	remaining: 1h 39m 9s
4015:	learn: 0.2038155	total: 4m 8s	remaining: 1h 39m 8s
4016:	learn: 0.2037909	total: 4m 8s	remaining: 1h 39m 8s
4017:	learn: 0.2037674	total: 4m 9s	remaining: 1h 39m 8s
4018:	learn: 0.2037559	total: 4m 9s	remaining: 1h 39m 8s
4019:	learn: 0.2037473	total: 4m 9s	remaining: 1h 39m 8s
4020:	learn: 0.2037248	total: 4m 9s	remaining: 1h 39m 8s
4021:	learn: 0.2037130	total: 4m 9s	remaining: 1h 39m 8s
4022:	learn: 0.2037055	total: 4m 9s	remaining: 1h 39m 8s
4023:	learn: 0.2036896	total: 4

4149:	learn: 0.2017538	total: 4m 17s	remaining: 1h 39m 1s
4150:	learn: 0.2017360	total: 4m 17s	remaining: 1h 39m 1s
4151:	learn: 0.2017142	total: 4m 17s	remaining: 1h 39m 1s
4152:	learn: 0.2016975	total: 4m 17s	remaining: 1h 39m 1s
4153:	learn: 0.2016865	total: 4m 17s	remaining: 1h 39m 1s
4154:	learn: 0.2016755	total: 4m 17s	remaining: 1h 39m 1s
4155:	learn: 0.2016638	total: 4m 17s	remaining: 1h 39m 1s
4156:	learn: 0.2016408	total: 4m 17s	remaining: 1h 39m
4157:	learn: 0.2016351	total: 4m 17s	remaining: 1h 39m
4158:	learn: 0.2016250	total: 4m 17s	remaining: 1h 39m
4159:	learn: 0.2016143	total: 4m 17s	remaining: 1h 39m
4160:	learn: 0.2015948	total: 4m 17s	remaining: 1h 39m
4161:	learn: 0.2015824	total: 4m 17s	remaining: 1h 39m
4162:	learn: 0.2015682	total: 4m 18s	remaining: 1h 39m
4163:	learn: 0.2015603	total: 4m 18s	remaining: 1h 39m
4164:	learn: 0.2015429	total: 4m 18s	remaining: 1h 39m
4165:	learn: 0.2015286	total: 4m 18s	remaining: 1h 39m
4166:	learn: 0.2015019	total: 4m 18s	remaini

4293:	learn: 0.1997082	total: 4m 26s	remaining: 1h 38m 53s
4294:	learn: 0.1996885	total: 4m 26s	remaining: 1h 38m 53s
4295:	learn: 0.1996809	total: 4m 26s	remaining: 1h 38m 53s
4296:	learn: 0.1996646	total: 4m 26s	remaining: 1h 38m 53s
4297:	learn: 0.1996587	total: 4m 26s	remaining: 1h 38m 53s
4298:	learn: 0.1996455	total: 4m 26s	remaining: 1h 38m 53s
4299:	learn: 0.1996399	total: 4m 26s	remaining: 1h 38m 53s
4300:	learn: 0.1996269	total: 4m 26s	remaining: 1h 38m 53s
4301:	learn: 0.1996130	total: 4m 26s	remaining: 1h 38m 53s
4302:	learn: 0.1995976	total: 4m 26s	remaining: 1h 38m 53s
4303:	learn: 0.1995765	total: 4m 26s	remaining: 1h 38m 52s
4304:	learn: 0.1995684	total: 4m 26s	remaining: 1h 38m 52s
4305:	learn: 0.1995592	total: 4m 26s	remaining: 1h 38m 52s
4306:	learn: 0.1995476	total: 4m 27s	remaining: 1h 38m 52s
4307:	learn: 0.1995413	total: 4m 27s	remaining: 1h 38m 52s
4308:	learn: 0.1995232	total: 4m 27s	remaining: 1h 38m 52s
4309:	learn: 0.1995148	total: 4m 27s	remaining: 1h 38m 5

4433:	learn: 0.1978077	total: 4m 34s	remaining: 1h 38m 43s
4434:	learn: 0.1978019	total: 4m 34s	remaining: 1h 38m 43s
4435:	learn: 0.1977886	total: 4m 34s	remaining: 1h 38m 43s
4436:	learn: 0.1977810	total: 4m 35s	remaining: 1h 38m 43s
4437:	learn: 0.1977675	total: 4m 35s	remaining: 1h 38m 43s
4438:	learn: 0.1977624	total: 4m 35s	remaining: 1h 38m 43s
4439:	learn: 0.1977539	total: 4m 35s	remaining: 1h 38m 43s
4440:	learn: 0.1977415	total: 4m 35s	remaining: 1h 38m 43s
4441:	learn: 0.1977362	total: 4m 35s	remaining: 1h 38m 43s
4442:	learn: 0.1977256	total: 4m 35s	remaining: 1h 38m 43s
4443:	learn: 0.1977152	total: 4m 35s	remaining: 1h 38m 43s
4444:	learn: 0.1977053	total: 4m 35s	remaining: 1h 38m 43s
4445:	learn: 0.1977010	total: 4m 35s	remaining: 1h 38m 43s
4446:	learn: 0.1976948	total: 4m 35s	remaining: 1h 38m 43s
4447:	learn: 0.1976792	total: 4m 35s	remaining: 1h 38m 43s
4448:	learn: 0.1976613	total: 4m 35s	remaining: 1h 38m 43s
4449:	learn: 0.1976457	total: 4m 35s	remaining: 1h 38m 4

4573:	learn: 0.1960067	total: 4m 43s	remaining: 1h 38m 35s
4574:	learn: 0.1960037	total: 4m 43s	remaining: 1h 38m 35s
4575:	learn: 0.1959953	total: 4m 43s	remaining: 1h 38m 35s
4576:	learn: 0.1959811	total: 4m 43s	remaining: 1h 38m 35s
4577:	learn: 0.1959655	total: 4m 43s	remaining: 1h 38m 35s
4578:	learn: 0.1959554	total: 4m 43s	remaining: 1h 38m 35s
4579:	learn: 0.1959432	total: 4m 43s	remaining: 1h 38m 35s
4580:	learn: 0.1959385	total: 4m 44s	remaining: 1h 38m 35s
4581:	learn: 0.1959302	total: 4m 44s	remaining: 1h 38m 35s
4582:	learn: 0.1959218	total: 4m 44s	remaining: 1h 38m 35s
4583:	learn: 0.1959129	total: 4m 44s	remaining: 1h 38m 35s
4584:	learn: 0.1959024	total: 4m 44s	remaining: 1h 38m 35s
4585:	learn: 0.1958834	total: 4m 44s	remaining: 1h 38m 35s
4586:	learn: 0.1958755	total: 4m 44s	remaining: 1h 38m 35s
4587:	learn: 0.1958565	total: 4m 44s	remaining: 1h 38m 35s
4588:	learn: 0.1958503	total: 4m 44s	remaining: 1h 38m 35s
4589:	learn: 0.1958421	total: 4m 44s	remaining: 1h 38m 3

4712:	learn: 0.1944119	total: 4m 52s	remaining: 1h 38m 26s
4713:	learn: 0.1943975	total: 4m 52s	remaining: 1h 38m 26s
4714:	learn: 0.1943777	total: 4m 52s	remaining: 1h 38m 26s
4715:	learn: 0.1943614	total: 4m 52s	remaining: 1h 38m 26s
4716:	learn: 0.1943453	total: 4m 52s	remaining: 1h 38m 26s
4717:	learn: 0.1943320	total: 4m 52s	remaining: 1h 38m 26s
4718:	learn: 0.1943146	total: 4m 52s	remaining: 1h 38m 26s
4719:	learn: 0.1942984	total: 4m 52s	remaining: 1h 38m 26s
4720:	learn: 0.1942790	total: 4m 52s	remaining: 1h 38m 26s
4721:	learn: 0.1942710	total: 4m 52s	remaining: 1h 38m 26s
4722:	learn: 0.1942588	total: 4m 52s	remaining: 1h 38m 25s
4723:	learn: 0.1942446	total: 4m 52s	remaining: 1h 38m 25s
4724:	learn: 0.1942386	total: 4m 52s	remaining: 1h 38m 25s
4725:	learn: 0.1942236	total: 4m 52s	remaining: 1h 38m 25s
4726:	learn: 0.1942114	total: 4m 53s	remaining: 1h 38m 25s
4727:	learn: 0.1942003	total: 4m 53s	remaining: 1h 38m 25s
4728:	learn: 0.1941858	total: 4m 53s	remaining: 1h 38m 2

4852:	learn: 0.1928137	total: 5m	remaining: 1h 38m 18s
4853:	learn: 0.1927976	total: 5m	remaining: 1h 38m 18s
4854:	learn: 0.1927904	total: 5m	remaining: 1h 38m 18s
4855:	learn: 0.1927785	total: 5m 1s	remaining: 1h 38m 17s
4856:	learn: 0.1927715	total: 5m 1s	remaining: 1h 38m 17s
4857:	learn: 0.1927633	total: 5m 1s	remaining: 1h 38m 17s
4858:	learn: 0.1927437	total: 5m 1s	remaining: 1h 38m 17s
4859:	learn: 0.1927341	total: 5m 1s	remaining: 1h 38m 17s
4860:	learn: 0.1927180	total: 5m 1s	remaining: 1h 38m 17s
4861:	learn: 0.1926977	total: 5m 1s	remaining: 1h 38m 17s
4862:	learn: 0.1926896	total: 5m 1s	remaining: 1h 38m 17s
4863:	learn: 0.1926702	total: 5m 1s	remaining: 1h 38m 17s
4864:	learn: 0.1926551	total: 5m 1s	remaining: 1h 38m 17s
4865:	learn: 0.1926464	total: 5m 1s	remaining: 1h 38m 17s
4866:	learn: 0.1926347	total: 5m 1s	remaining: 1h 38m 17s
4867:	learn: 0.1926266	total: 5m 1s	remaining: 1h 38m 17s
4868:	learn: 0.1926205	total: 5m 1s	remaining: 1h 38m 16s
4869:	learn: 0.1926139	

4996:	learn: 0.1913466	total: 5m 9s	remaining: 1h 38m 9s
4997:	learn: 0.1913274	total: 5m 9s	remaining: 1h 38m 9s
4998:	learn: 0.1913208	total: 5m 9s	remaining: 1h 38m 9s
4999:	learn: 0.1913128	total: 5m 9s	remaining: 1h 38m 9s
5000:	learn: 0.1913068	total: 5m 10s	remaining: 1h 38m 8s
5001:	learn: 0.1912960	total: 5m 10s	remaining: 1h 38m 8s
5002:	learn: 0.1912812	total: 5m 10s	remaining: 1h 38m 8s
5003:	learn: 0.1912754	total: 5m 10s	remaining: 1h 38m 8s
5004:	learn: 0.1912635	total: 5m 10s	remaining: 1h 38m 8s
5005:	learn: 0.1912580	total: 5m 10s	remaining: 1h 38m 8s
5006:	learn: 0.1912511	total: 5m 10s	remaining: 1h 38m 8s
5007:	learn: 0.1912386	total: 5m 10s	remaining: 1h 38m 8s
5008:	learn: 0.1912282	total: 5m 10s	remaining: 1h 38m 8s
5009:	learn: 0.1912159	total: 5m 10s	remaining: 1h 38m 8s
5010:	learn: 0.1912085	total: 5m 10s	remaining: 1h 38m 8s
5011:	learn: 0.1912031	total: 5m 10s	remaining: 1h 38m 7s
5012:	learn: 0.1911863	total: 5m 10s	remaining: 1h 38m 7s
5013:	learn: 0.191

5140:	learn: 0.1899908	total: 5m 18s	remaining: 1h 38m 1s
5141:	learn: 0.1899790	total: 5m 18s	remaining: 1h 38m 1s
5142:	learn: 0.1899674	total: 5m 18s	remaining: 1h 38m
5143:	learn: 0.1899524	total: 5m 18s	remaining: 1h 38m
5144:	learn: 0.1899452	total: 5m 18s	remaining: 1h 38m
5145:	learn: 0.1899371	total: 5m 19s	remaining: 1h 38m
5146:	learn: 0.1899198	total: 5m 19s	remaining: 1h 38m
5147:	learn: 0.1899173	total: 5m 19s	remaining: 1h 38m
5148:	learn: 0.1899118	total: 5m 19s	remaining: 1h 38m
5149:	learn: 0.1899049	total: 5m 19s	remaining: 1h 38m
5150:	learn: 0.1898979	total: 5m 19s	remaining: 1h 38m
5151:	learn: 0.1898849	total: 5m 19s	remaining: 1h 38m
5152:	learn: 0.1898767	total: 5m 19s	remaining: 1h 38m
5153:	learn: 0.1898700	total: 5m 19s	remaining: 1h 38m
5154:	learn: 0.1898654	total: 5m 19s	remaining: 1h 38m
5155:	learn: 0.1898541	total: 5m 19s	remaining: 1h 38m
5156:	learn: 0.1898425	total: 5m 19s	remaining: 1h 38m
5157:	learn: 0.1898298	total: 5m 19s	remaining: 1h 38m
5158

5283:	learn: 0.1887318	total: 5m 27s	remaining: 1h 37m 52s
5284:	learn: 0.1887270	total: 5m 27s	remaining: 1h 37m 52s
5285:	learn: 0.1887155	total: 5m 27s	remaining: 1h 37m 52s
5286:	learn: 0.1887117	total: 5m 27s	remaining: 1h 37m 52s
5287:	learn: 0.1886949	total: 5m 27s	remaining: 1h 37m 52s
5288:	learn: 0.1886857	total: 5m 27s	remaining: 1h 37m 52s
5289:	learn: 0.1886807	total: 5m 28s	remaining: 1h 37m 52s
5290:	learn: 0.1886761	total: 5m 28s	remaining: 1h 37m 52s
5291:	learn: 0.1886697	total: 5m 28s	remaining: 1h 37m 52s
5292:	learn: 0.1886511	total: 5m 28s	remaining: 1h 37m 52s
5293:	learn: 0.1886403	total: 5m 28s	remaining: 1h 37m 52s
5294:	learn: 0.1886368	total: 5m 28s	remaining: 1h 37m 52s
5295:	learn: 0.1886301	total: 5m 28s	remaining: 1h 37m 52s
5296:	learn: 0.1886250	total: 5m 28s	remaining: 1h 37m 52s
5297:	learn: 0.1886119	total: 5m 28s	remaining: 1h 37m 52s
5298:	learn: 0.1885971	total: 5m 28s	remaining: 1h 37m 51s
5299:	learn: 0.1885896	total: 5m 28s	remaining: 1h 37m 5

5422:	learn: 0.1875143	total: 5m 36s	remaining: 1h 37m 43s
5423:	learn: 0.1875058	total: 5m 36s	remaining: 1h 37m 43s
5424:	learn: 0.1874959	total: 5m 36s	remaining: 1h 37m 43s
5425:	learn: 0.1874872	total: 5m 36s	remaining: 1h 37m 43s
5426:	learn: 0.1874789	total: 5m 36s	remaining: 1h 37m 43s
5427:	learn: 0.1874716	total: 5m 36s	remaining: 1h 37m 43s
5428:	learn: 0.1874661	total: 5m 36s	remaining: 1h 37m 43s
5429:	learn: 0.1874503	total: 5m 36s	remaining: 1h 37m 43s
5430:	learn: 0.1874445	total: 5m 36s	remaining: 1h 37m 43s
5431:	learn: 0.1874394	total: 5m 36s	remaining: 1h 37m 42s
5432:	learn: 0.1874357	total: 5m 36s	remaining: 1h 37m 42s
5433:	learn: 0.1874258	total: 5m 36s	remaining: 1h 37m 42s
5434:	learn: 0.1874224	total: 5m 36s	remaining: 1h 37m 42s
5435:	learn: 0.1874082	total: 5m 37s	remaining: 1h 37m 42s
5436:	learn: 0.1873993	total: 5m 37s	remaining: 1h 37m 42s
5437:	learn: 0.1873959	total: 5m 37s	remaining: 1h 37m 42s
5438:	learn: 0.1873839	total: 5m 37s	remaining: 1h 37m 4

5562:	learn: 0.1862402	total: 5m 44s	remaining: 1h 37m 33s
5563:	learn: 0.1862277	total: 5m 44s	remaining: 1h 37m 33s
5564:	learn: 0.1862105	total: 5m 44s	remaining: 1h 37m 33s
5565:	learn: 0.1862003	total: 5m 45s	remaining: 1h 37m 33s
5566:	learn: 0.1861914	total: 5m 45s	remaining: 1h 37m 33s
5567:	learn: 0.1861776	total: 5m 45s	remaining: 1h 37m 33s
5568:	learn: 0.1861664	total: 5m 45s	remaining: 1h 37m 33s
5569:	learn: 0.1861624	total: 5m 45s	remaining: 1h 37m 33s
5570:	learn: 0.1861531	total: 5m 45s	remaining: 1h 37m 33s
5571:	learn: 0.1861458	total: 5m 45s	remaining: 1h 37m 33s
5572:	learn: 0.1861399	total: 5m 45s	remaining: 1h 37m 33s
5573:	learn: 0.1861342	total: 5m 45s	remaining: 1h 37m 32s
5574:	learn: 0.1861300	total: 5m 45s	remaining: 1h 37m 32s
5575:	learn: 0.1861257	total: 5m 45s	remaining: 1h 37m 32s
5576:	learn: 0.1861168	total: 5m 45s	remaining: 1h 37m 32s
5577:	learn: 0.1861130	total: 5m 45s	remaining: 1h 37m 32s
5578:	learn: 0.1861081	total: 5m 45s	remaining: 1h 37m 3

5701:	learn: 0.1851628	total: 5m 53s	remaining: 1h 37m 25s
5702:	learn: 0.1851518	total: 5m 53s	remaining: 1h 37m 25s
5703:	learn: 0.1851488	total: 5m 53s	remaining: 1h 37m 25s
5704:	learn: 0.1851420	total: 5m 53s	remaining: 1h 37m 25s
5705:	learn: 0.1851332	total: 5m 53s	remaining: 1h 37m 25s
5706:	learn: 0.1851303	total: 5m 53s	remaining: 1h 37m 25s
5707:	learn: 0.1851259	total: 5m 53s	remaining: 1h 37m 25s
5708:	learn: 0.1851214	total: 5m 53s	remaining: 1h 37m 25s
5709:	learn: 0.1851094	total: 5m 53s	remaining: 1h 37m 25s
5710:	learn: 0.1851055	total: 5m 54s	remaining: 1h 37m 25s
5711:	learn: 0.1850943	total: 5m 54s	remaining: 1h 37m 25s
5712:	learn: 0.1850872	total: 5m 54s	remaining: 1h 37m 25s
5713:	learn: 0.1850755	total: 5m 54s	remaining: 1h 37m 25s
5714:	learn: 0.1850659	total: 5m 54s	remaining: 1h 37m 25s
5715:	learn: 0.1850497	total: 5m 54s	remaining: 1h 37m 25s
5716:	learn: 0.1850388	total: 5m 54s	remaining: 1h 37m 25s
5717:	learn: 0.1850313	total: 5m 54s	remaining: 1h 37m 2

5845:	learn: 0.1840203	total: 6m 2s	remaining: 1h 37m 17s
5846:	learn: 0.1840179	total: 6m 2s	remaining: 1h 37m 17s
5847:	learn: 0.1840099	total: 6m 2s	remaining: 1h 37m 17s
5848:	learn: 0.1840018	total: 6m 2s	remaining: 1h 37m 17s
5849:	learn: 0.1839927	total: 6m 2s	remaining: 1h 37m 17s
5850:	learn: 0.1839879	total: 6m 2s	remaining: 1h 37m 17s
5851:	learn: 0.1839853	total: 6m 2s	remaining: 1h 37m 17s
5852:	learn: 0.1839829	total: 6m 2s	remaining: 1h 37m 17s
5853:	learn: 0.1839727	total: 6m 2s	remaining: 1h 37m 17s
5854:	learn: 0.1839606	total: 6m 3s	remaining: 1h 37m 17s
5855:	learn: 0.1839530	total: 6m 3s	remaining: 1h 37m 17s
5856:	learn: 0.1839371	total: 6m 3s	remaining: 1h 37m 17s
5857:	learn: 0.1839256	total: 6m 3s	remaining: 1h 37m 17s
5858:	learn: 0.1839156	total: 6m 3s	remaining: 1h 37m 17s
5859:	learn: 0.1839114	total: 6m 3s	remaining: 1h 37m 17s
5860:	learn: 0.1839090	total: 6m 3s	remaining: 1h 37m 17s
5861:	learn: 0.1839047	total: 6m 3s	remaining: 1h 37m 17s
5862:	learn: 0

5989:	learn: 0.1828862	total: 6m 11s	remaining: 1h 37m 10s
5990:	learn: 0.1828828	total: 6m 11s	remaining: 1h 37m 10s
5991:	learn: 0.1828751	total: 6m 11s	remaining: 1h 37m 10s
5992:	learn: 0.1828620	total: 6m 11s	remaining: 1h 37m 10s
5993:	learn: 0.1828551	total: 6m 11s	remaining: 1h 37m 10s
5994:	learn: 0.1828502	total: 6m 11s	remaining: 1h 37m 10s
5995:	learn: 0.1828353	total: 6m 11s	remaining: 1h 37m 10s
5996:	learn: 0.1828254	total: 6m 11s	remaining: 1h 37m 10s
5997:	learn: 0.1828159	total: 6m 11s	remaining: 1h 37m 10s
5998:	learn: 0.1828033	total: 6m 12s	remaining: 1h 37m 9s
5999:	learn: 0.1828001	total: 6m 12s	remaining: 1h 37m 9s
6000:	learn: 0.1827969	total: 6m 12s	remaining: 1h 37m 9s
6001:	learn: 0.1827856	total: 6m 12s	remaining: 1h 37m 9s
6002:	learn: 0.1827774	total: 6m 12s	remaining: 1h 37m 9s
6003:	learn: 0.1827648	total: 6m 12s	remaining: 1h 37m 9s
6004:	learn: 0.1827625	total: 6m 12s	remaining: 1h 37m 9s
6005:	learn: 0.1827482	total: 6m 12s	remaining: 1h 37m 9s
6006:

6133:	learn: 0.1818510	total: 6m 20s	remaining: 1h 37m 1s
6134:	learn: 0.1818452	total: 6m 20s	remaining: 1h 37m 1s
6135:	learn: 0.1818317	total: 6m 20s	remaining: 1h 37m 1s
6136:	learn: 0.1818225	total: 6m 20s	remaining: 1h 37m 1s
6137:	learn: 0.1818122	total: 6m 20s	remaining: 1h 37m 1s
6138:	learn: 0.1818063	total: 6m 20s	remaining: 1h 37m 1s
6139:	learn: 0.1818024	total: 6m 20s	remaining: 1h 37m 1s
6140:	learn: 0.1817888	total: 6m 20s	remaining: 1h 37m 1s
6141:	learn: 0.1817864	total: 6m 20s	remaining: 1h 37m
6142:	learn: 0.1817783	total: 6m 20s	remaining: 1h 37m
6143:	learn: 0.1817747	total: 6m 21s	remaining: 1h 37m
6144:	learn: 0.1817637	total: 6m 21s	remaining: 1h 37m
6145:	learn: 0.1817600	total: 6m 21s	remaining: 1h 37m
6146:	learn: 0.1817532	total: 6m 21s	remaining: 1h 37m
6147:	learn: 0.1817478	total: 6m 21s	remaining: 1h 37m
6148:	learn: 0.1817351	total: 6m 21s	remaining: 1h 37m
6149:	learn: 0.1817323	total: 6m 21s	remaining: 1h 37m
6150:	learn: 0.1817302	total: 6m 21s	rema

6277:	learn: 0.1808554	total: 6m 29s	remaining: 1h 36m 54s
6278:	learn: 0.1808476	total: 6m 29s	remaining: 1h 36m 54s
6279:	learn: 0.1808435	total: 6m 29s	remaining: 1h 36m 53s
6280:	learn: 0.1808412	total: 6m 29s	remaining: 1h 36m 53s
6281:	learn: 0.1808303	total: 6m 29s	remaining: 1h 36m 53s
6282:	learn: 0.1808244	total: 6m 29s	remaining: 1h 36m 53s
6283:	learn: 0.1808219	total: 6m 29s	remaining: 1h 36m 53s
6284:	learn: 0.1808173	total: 6m 29s	remaining: 1h 36m 53s
6285:	learn: 0.1808131	total: 6m 29s	remaining: 1h 36m 53s
6286:	learn: 0.1808095	total: 6m 30s	remaining: 1h 36m 53s
6287:	learn: 0.1808064	total: 6m 30s	remaining: 1h 36m 53s
6288:	learn: 0.1807992	total: 6m 30s	remaining: 1h 36m 53s
6289:	learn: 0.1807874	total: 6m 30s	remaining: 1h 36m 53s
6290:	learn: 0.1807836	total: 6m 30s	remaining: 1h 36m 53s
6291:	learn: 0.1807805	total: 6m 30s	remaining: 1h 36m 53s
6292:	learn: 0.1807754	total: 6m 30s	remaining: 1h 36m 53s
6293:	learn: 0.1807727	total: 6m 30s	remaining: 1h 36m 5

6417:	learn: 0.1799299	total: 6m 38s	remaining: 1h 36m 45s
6418:	learn: 0.1799270	total: 6m 38s	remaining: 1h 36m 45s
6419:	learn: 0.1799232	total: 6m 38s	remaining: 1h 36m 45s
6420:	learn: 0.1799130	total: 6m 38s	remaining: 1h 36m 45s
6421:	learn: 0.1799103	total: 6m 38s	remaining: 1h 36m 45s
6422:	learn: 0.1798954	total: 6m 38s	remaining: 1h 36m 45s
6423:	learn: 0.1798942	total: 6m 38s	remaining: 1h 36m 45s
6424:	learn: 0.1798899	total: 6m 38s	remaining: 1h 36m 45s
6425:	learn: 0.1798866	total: 6m 38s	remaining: 1h 36m 45s
6426:	learn: 0.1798813	total: 6m 38s	remaining: 1h 36m 45s
6427:	learn: 0.1798758	total: 6m 38s	remaining: 1h 36m 45s
6428:	learn: 0.1798702	total: 6m 38s	remaining: 1h 36m 45s
6429:	learn: 0.1798583	total: 6m 38s	remaining: 1h 36m 45s
6430:	learn: 0.1798514	total: 6m 38s	remaining: 1h 36m 45s
6431:	learn: 0.1798461	total: 6m 39s	remaining: 1h 36m 45s
6432:	learn: 0.1798430	total: 6m 39s	remaining: 1h 36m 45s
6433:	learn: 0.1798330	total: 6m 39s	remaining: 1h 36m 4

6557:	learn: 0.1790847	total: 6m 46s	remaining: 1h 36m 36s
6558:	learn: 0.1790826	total: 6m 46s	remaining: 1h 36m 36s
6559:	learn: 0.1790720	total: 6m 46s	remaining: 1h 36m 36s
6560:	learn: 0.1790685	total: 6m 47s	remaining: 1h 36m 36s
6561:	learn: 0.1790578	total: 6m 47s	remaining: 1h 36m 36s
6562:	learn: 0.1790544	total: 6m 47s	remaining: 1h 36m 36s
6563:	learn: 0.1790485	total: 6m 47s	remaining: 1h 36m 36s
6564:	learn: 0.1790375	total: 6m 47s	remaining: 1h 36m 36s
6565:	learn: 0.1790358	total: 6m 47s	remaining: 1h 36m 36s
6566:	learn: 0.1790268	total: 6m 47s	remaining: 1h 36m 36s
6567:	learn: 0.1790126	total: 6m 47s	remaining: 1h 36m 36s
6568:	learn: 0.1790075	total: 6m 47s	remaining: 1h 36m 36s
6569:	learn: 0.1789969	total: 6m 47s	remaining: 1h 36m 35s
6570:	learn: 0.1789867	total: 6m 47s	remaining: 1h 36m 35s
6571:	learn: 0.1789774	total: 6m 47s	remaining: 1h 36m 35s
6572:	learn: 0.1789673	total: 6m 47s	remaining: 1h 36m 35s
6573:	learn: 0.1789650	total: 6m 47s	remaining: 1h 36m 3

6697:	learn: 0.1782267	total: 6m 55s	remaining: 1h 36m 27s
6698:	learn: 0.1782243	total: 6m 55s	remaining: 1h 36m 27s
6699:	learn: 0.1782126	total: 6m 55s	remaining: 1h 36m 27s
6700:	learn: 0.1782112	total: 6m 55s	remaining: 1h 36m 27s
6701:	learn: 0.1782066	total: 6m 55s	remaining: 1h 36m 27s
6702:	learn: 0.1782017	total: 6m 55s	remaining: 1h 36m 27s
6703:	learn: 0.1781932	total: 6m 55s	remaining: 1h 36m 27s
6704:	learn: 0.1781819	total: 6m 55s	remaining: 1h 36m 27s
6705:	learn: 0.1781771	total: 6m 55s	remaining: 1h 36m 27s
6706:	learn: 0.1781735	total: 6m 56s	remaining: 1h 36m 27s
6707:	learn: 0.1781700	total: 6m 56s	remaining: 1h 36m 27s
6708:	learn: 0.1781682	total: 6m 56s	remaining: 1h 36m 27s
6709:	learn: 0.1781655	total: 6m 56s	remaining: 1h 36m 27s
6710:	learn: 0.1781560	total: 6m 56s	remaining: 1h 36m 27s
6711:	learn: 0.1781517	total: 6m 56s	remaining: 1h 36m 27s
6712:	learn: 0.1781445	total: 6m 56s	remaining: 1h 36m 27s
6713:	learn: 0.1781361	total: 6m 56s	remaining: 1h 36m 2

6841:	learn: 0.1774588	total: 7m 4s	remaining: 1h 36m 18s
6842:	learn: 0.1774515	total: 7m 4s	remaining: 1h 36m 18s
6843:	learn: 0.1774432	total: 7m 4s	remaining: 1h 36m 18s
6844:	learn: 0.1774365	total: 7m 4s	remaining: 1h 36m 18s
6845:	learn: 0.1774290	total: 7m 4s	remaining: 1h 36m 18s
6846:	learn: 0.1774223	total: 7m 4s	remaining: 1h 36m 18s
6847:	learn: 0.1774172	total: 7m 4s	remaining: 1h 36m 18s
6848:	learn: 0.1774071	total: 7m 4s	remaining: 1h 36m 18s
6849:	learn: 0.1774045	total: 7m 4s	remaining: 1h 36m 18s
6850:	learn: 0.1774007	total: 7m 4s	remaining: 1h 36m 18s
6851:	learn: 0.1773983	total: 7m 5s	remaining: 1h 36m 18s
6852:	learn: 0.1773843	total: 7m 5s	remaining: 1h 36m 18s
6853:	learn: 0.1773757	total: 7m 5s	remaining: 1h 36m 18s
6854:	learn: 0.1773720	total: 7m 5s	remaining: 1h 36m 18s
6855:	learn: 0.1773706	total: 7m 5s	remaining: 1h 36m 17s
6856:	learn: 0.1773629	total: 7m 5s	remaining: 1h 36m 17s
6857:	learn: 0.1773570	total: 7m 5s	remaining: 1h 36m 17s
6858:	learn: 0

6985:	learn: 0.1766259	total: 7m 13s	remaining: 1h 36m 9s
6986:	learn: 0.1766174	total: 7m 13s	remaining: 1h 36m 9s
6987:	learn: 0.1766114	total: 7m 13s	remaining: 1h 36m 9s
6988:	learn: 0.1766093	total: 7m 13s	remaining: 1h 36m 9s
6989:	learn: 0.1766068	total: 7m 13s	remaining: 1h 36m 9s
6990:	learn: 0.1766029	total: 7m 13s	remaining: 1h 36m 9s
6991:	learn: 0.1765995	total: 7m 13s	remaining: 1h 36m 9s
6992:	learn: 0.1765894	total: 7m 13s	remaining: 1h 36m 9s
6993:	learn: 0.1765815	total: 7m 13s	remaining: 1h 36m 9s
6994:	learn: 0.1765745	total: 7m 13s	remaining: 1h 36m 9s
6995:	learn: 0.1765729	total: 7m 14s	remaining: 1h 36m 9s
6996:	learn: 0.1765634	total: 7m 14s	remaining: 1h 36m 9s
6997:	learn: 0.1765555	total: 7m 14s	remaining: 1h 36m 9s
6998:	learn: 0.1765534	total: 7m 14s	remaining: 1h 36m 9s
6999:	learn: 0.1765491	total: 7m 14s	remaining: 1h 36m 9s
7000:	learn: 0.1765440	total: 7m 14s	remaining: 1h 36m 9s
7001:	learn: 0.1765419	total: 7m 14s	remaining: 1h 36m 9s
7002:	learn: 0

7129:	learn: 0.1758941	total: 7m 22s	remaining: 1h 36m 1s
7130:	learn: 0.1758925	total: 7m 22s	remaining: 1h 36m 1s
7131:	learn: 0.1758899	total: 7m 22s	remaining: 1h 36m 1s
7132:	learn: 0.1758778	total: 7m 22s	remaining: 1h 36m 1s
7133:	learn: 0.1758696	total: 7m 22s	remaining: 1h 36m 1s
7134:	learn: 0.1758628	total: 7m 22s	remaining: 1h 36m 1s
7135:	learn: 0.1758588	total: 7m 22s	remaining: 1h 36m 1s
7136:	learn: 0.1758533	total: 7m 22s	remaining: 1h 36m
7137:	learn: 0.1758506	total: 7m 22s	remaining: 1h 36m
7138:	learn: 0.1758431	total: 7m 22s	remaining: 1h 36m
7139:	learn: 0.1758410	total: 7m 22s	remaining: 1h 36m
7140:	learn: 0.1758400	total: 7m 23s	remaining: 1h 36m
7141:	learn: 0.1758364	total: 7m 23s	remaining: 1h 36m
7142:	learn: 0.1758342	total: 7m 23s	remaining: 1h 36m
7143:	learn: 0.1758243	total: 7m 23s	remaining: 1h 36m
7144:	learn: 0.1758220	total: 7m 23s	remaining: 1h 36m
7145:	learn: 0.1758172	total: 7m 23s	remaining: 1h 36m
7146:	learn: 0.1758096	total: 7m 23s	remaini

7273:	learn: 0.1751467	total: 7m 31s	remaining: 1h 35m 52s
7274:	learn: 0.1751432	total: 7m 31s	remaining: 1h 35m 52s
7275:	learn: 0.1751420	total: 7m 31s	remaining: 1h 35m 51s
7276:	learn: 0.1751314	total: 7m 31s	remaining: 1h 35m 51s
7277:	learn: 0.1751306	total: 7m 31s	remaining: 1h 35m 51s
7278:	learn: 0.1751281	total: 7m 31s	remaining: 1h 35m 51s
7279:	learn: 0.1751198	total: 7m 31s	remaining: 1h 35m 51s
7280:	learn: 0.1751101	total: 7m 31s	remaining: 1h 35m 51s
7281:	learn: 0.1751046	total: 7m 31s	remaining: 1h 35m 51s
7282:	learn: 0.1751013	total: 7m 31s	remaining: 1h 35m 51s
7283:	learn: 0.1750960	total: 7m 31s	remaining: 1h 35m 51s
7284:	learn: 0.1750836	total: 7m 31s	remaining: 1h 35m 51s
7285:	learn: 0.1750776	total: 7m 31s	remaining: 1h 35m 51s
7286:	learn: 0.1750762	total: 7m 32s	remaining: 1h 35m 51s
7287:	learn: 0.1750711	total: 7m 32s	remaining: 1h 35m 51s
7288:	learn: 0.1750630	total: 7m 32s	remaining: 1h 35m 51s
7289:	learn: 0.1750562	total: 7m 32s	remaining: 1h 35m 5

7413:	learn: 0.1743831	total: 7m 39s	remaining: 1h 35m 43s
7414:	learn: 0.1743811	total: 7m 39s	remaining: 1h 35m 43s
7415:	learn: 0.1743719	total: 7m 40s	remaining: 1h 35m 42s
7416:	learn: 0.1743703	total: 7m 40s	remaining: 1h 35m 42s
7417:	learn: 0.1743669	total: 7m 40s	remaining: 1h 35m 42s
7418:	learn: 0.1743620	total: 7m 40s	remaining: 1h 35m 42s
7419:	learn: 0.1743499	total: 7m 40s	remaining: 1h 35m 42s
7420:	learn: 0.1743387	total: 7m 40s	remaining: 1h 35m 42s
7421:	learn: 0.1743356	total: 7m 40s	remaining: 1h 35m 42s
7422:	learn: 0.1743327	total: 7m 40s	remaining: 1h 35m 42s
7423:	learn: 0.1743306	total: 7m 40s	remaining: 1h 35m 42s
7424:	learn: 0.1743295	total: 7m 40s	remaining: 1h 35m 42s
7425:	learn: 0.1743287	total: 7m 40s	remaining: 1h 35m 42s
7426:	learn: 0.1743243	total: 7m 40s	remaining: 1h 35m 42s
7427:	learn: 0.1743120	total: 7m 40s	remaining: 1h 35m 42s
7428:	learn: 0.1743048	total: 7m 40s	remaining: 1h 35m 42s
7429:	learn: 0.1742949	total: 7m 40s	remaining: 1h 35m 4

7553:	learn: 0.1736999	total: 7m 48s	remaining: 1h 35m 34s
7554:	learn: 0.1736944	total: 7m 48s	remaining: 1h 35m 34s
7555:	learn: 0.1736881	total: 7m 48s	remaining: 1h 35m 34s
7556:	learn: 0.1736814	total: 7m 48s	remaining: 1h 35m 34s
7557:	learn: 0.1736730	total: 7m 48s	remaining: 1h 35m 34s
7558:	learn: 0.1736679	total: 7m 48s	remaining: 1h 35m 34s
7559:	learn: 0.1736649	total: 7m 48s	remaining: 1h 35m 34s
7560:	learn: 0.1736621	total: 7m 49s	remaining: 1h 35m 34s
7561:	learn: 0.1736545	total: 7m 49s	remaining: 1h 35m 34s
7562:	learn: 0.1736472	total: 7m 49s	remaining: 1h 35m 34s
7563:	learn: 0.1736401	total: 7m 49s	remaining: 1h 35m 34s
7564:	learn: 0.1736359	total: 7m 49s	remaining: 1h 35m 34s
7565:	learn: 0.1736272	total: 7m 49s	remaining: 1h 35m 33s
7566:	learn: 0.1736154	total: 7m 49s	remaining: 1h 35m 33s
7567:	learn: 0.1736085	total: 7m 49s	remaining: 1h 35m 33s
7568:	learn: 0.1736014	total: 7m 49s	remaining: 1h 35m 33s
7569:	learn: 0.1735973	total: 7m 49s	remaining: 1h 35m 3

7693:	learn: 0.1729915	total: 7m 57s	remaining: 1h 35m 25s
7694:	learn: 0.1729829	total: 7m 57s	remaining: 1h 35m 25s
7695:	learn: 0.1729786	total: 7m 57s	remaining: 1h 35m 25s
7696:	learn: 0.1729725	total: 7m 57s	remaining: 1h 35m 25s
7697:	learn: 0.1729701	total: 7m 57s	remaining: 1h 35m 25s
7698:	learn: 0.1729690	total: 7m 57s	remaining: 1h 35m 25s
7699:	learn: 0.1729665	total: 7m 57s	remaining: 1h 35m 25s
7700:	learn: 0.1729566	total: 7m 57s	remaining: 1h 35m 25s
7701:	learn: 0.1729495	total: 7m 57s	remaining: 1h 35m 24s
7702:	learn: 0.1729463	total: 7m 57s	remaining: 1h 35m 24s
7703:	learn: 0.1729365	total: 7m 57s	remaining: 1h 35m 24s
7704:	learn: 0.1729342	total: 7m 57s	remaining: 1h 35m 24s
7705:	learn: 0.1729314	total: 7m 57s	remaining: 1h 35m 24s
7706:	learn: 0.1729275	total: 7m 58s	remaining: 1h 35m 24s
7707:	learn: 0.1729246	total: 7m 58s	remaining: 1h 35m 24s
7708:	learn: 0.1729207	total: 7m 58s	remaining: 1h 35m 24s
7709:	learn: 0.1729181	total: 7m 58s	remaining: 1h 35m 2

#Test Model

In [ ]:
predictions = model.predict(X_test)
predictions = [round(value) for value in predictions]
np.unique(predictions , return_counts=True) 

In [101]:
accuracy = accuracy_score(y_test, predictions)
f1_macro = f1_score(y_test, predictions, average='macro')
f1_binary = f1_score(y_test, predictions, average='binary')
print("Accuracy: %.2f%%" % (accuracy * 100.0))
print("Macro F1: %.2f%%" % (f1_macro * 100.0))
print("Binary F1: %.2f%%" % (f1_binary * 100.0))

Accuracy: 96.16%
Macro F1: 86.48%
Binary F1: 97.92%


## Results

### catboost with categorical features 100000 epochs 
Accuracy: 96.16%
Macro F1: 86.48%
Binary F1: 97.92%
Submission_score : 0.97375

### catboost without dropping any feature with 400000 epochs

Accuracy: 97.41%
Macro F1: 90.87%
Binary F1: 98.60%
Submission Score : 0.97817

### catboost without dropping any feature  with 100000 epochs

Accuracy: 97.35%
Macro F1: 90.63%
Binary F1: 98.56%
Submission Score : 0.97848

### catboost after normalizing to z score with 10000 epochs

Accuracy: 96.70%
Macro F1: 87.82%
Binary F1: 98.22%

### catboost without dropping any feature with 10000 epochs

Accuracy: 96.82%
Macro F1: 88.95%
Binary F1: 98.27%
Submission Score : 0.97788

# Save Predictions

In [ ]:
predictions_1 = model.predict(test_1)
# predictions_1 = [round(value) for value in y_pred_1]
np.unique(predictions_1 , return_counts=True)

In [ ]:
submission_df = pd.read_csv("predications_submission.csv", 
                            index_col="tripid")
submission_df['prediction'] = predictions_1

submission_df.to_csv('predictions/catboost_categorical_100000_epochs.csv')

In [ ]:
#  y_pred = pd.DataFrame({'prediction':predictions_1})
#  y_pred["prediction"] = y_pred["prediction"].astype(int)
#  y_pred['prediction'] = y_pred['prediction'].apply(lambda x: 0 if x < 0.5 else 1)
#  y_pred.to_csv("predictions/predications_catboost_6_features_3_100000epochs.csv")
# y_pred.to_csv("/content/drive/My Drive/UNI/SEM7/CS4622_Machine_Learning/Project_1/Fair_Classification/predictions/predications_catboost_6_features_3_400000epochs.csv")